tests/js/client/shell/shell-cluster-collection.js

```diff
@@ -989,6 +989,47 @@ function ClusterCollectionSuite () {
       }
     },
 
+    // regression test for https://github.com/arangodb/arangodb/pull/21071
+    testCreateWithSlowCurrentUpdate : function () {
+      let setFailAt;
+      let removeFailAt;
+      if (isServer) {
+        if (internal.debugCanUseFailAt()) {
+          setFailAt = internal.debugSetFailAt;
+          removeFailAt = internal.debugRemoveFailAt;
+        }
+      } else {
+        const arango = internal.arango;
+        const coordinatorEndpoint = arango.getEndpoint();
+        if (debugCanUseFailAt(coordinatorEndpoint)) {
+          setFailAt = failurePoint => debugSetFailAt(coordinatorEndpoint, failurePoint);
+          removeFailAt = failurePoint => debugRemoveFailAt(coordinatorEndpoint, failurePoint);
+        }
+      }
+      if (!setFailAt) {
+        console.info('Failure tests disabled, skipping...');
+        return;
+      }
+
+      const failurePoint = 'ClusterInfo::slowCurrentSyncer';
+      const cn = "UnitTestsClusterCrud";
+      try {
+        // delay updates for current in the syncer thread
+        setFailAt(failurePoint);
+        // create a collection
+        const c = db._create(cn, { numberOfShards: 1, replicationFactor: 1 });
+        const docs = [];
+        for (let i = 0; i < 100; ++i) {
+          docs.push({ _key: "test" + i });
+        }
+        // inserting a document fails if it needs to look up a shard and
+        // ClusterInfo/Current hasn't caught up yet
+        c.insert(docs);
```

Comment:
wondering what is actually tested here, as this test does not contain any assertions, and will happily silence any exceptions in its `finally` block.
I think the thing tested here is the `insert` after the collection creation, but there should at least be a code comment here to indicate that, if not additional test assertions.

Reply:
The finally block does not silence any exceptions; only a catch would do that. So if the insert throws, the test fails. I've appended the comment to hopefully make it a little clearer.

Reply:
Thanks, you are obviously right about `finally` vs. `catch`.